consumer.go

```diff
@@ -524,10 +524,20 @@ func (child *partitionConsumer) parseRecords(batch *RecordBatch) ([]*ConsumerMes
 	var messages []*ConsumerMessage
 	var incomplete bool
 	prelude := true
-	originalOffset := child.offset
+
+	// Compacted records can lie about their offset delta, but that can be
+	// detected comparing the last offset delta from the record batch with
+	// actual offset delta of the last record. The difference (skew) should be
+	// applied to all records in the batch.
```

Comment:
I'm not sure this is true? Isn't the calculated skew only guaranteed to be true for the final record in the batch? Depending on where the records got compacted out the skew could be arbitrarily distributed in the batch.

Reply:
You are right. This is not correct. The problem is most likely with v0.11 producer. Please take a look at #1040 for more details. I am closing this PR.